### ISSUE TITLE
Quantile Regression

### ISSUE THREAD

**Issue by tbarker5252** [facebook/prophet#1176]
I have daily time series where for every day I have a variable (but large) number of real-valued measurements. For example, on day 1 I might have 20k measurements, and on day 2 i'll have 23k, etc. For every day, I essentially have a distribution of data, and what I want is to be able to forecast the percentiles of this data. So on day 3, I would like, say, 9 predictions of 10th, 20th, ..., and 90th percentiles.

How would I go about doing this with Prophet?

**Comment by bletham** [comment]
I think what you'd want to do is compute the quantile for each day, and then use that number as the time series to be forecasted.

To get a bunch of quantiles, you could just forecast them all separately.

Forecasting them separately isn't really ideal because it wouldn't for instance guarantee that p20 > p10. There isn't any true multivariate functionality in Prophet that would allow for joint forecasting of all of the percentiles at a time, but for this particular application you try the approach I describe in this comment:
https://github.com/facebook/prophet/issues/49#issuecomment-430383681

Basically, if you assume one of these quantiles is easier to forecast than the others (p50 is probably easier to forecast than p10/p90), then you would first forecast p50, and then for each of the others (say, p10) you would include the p50 forecast as an extra regressor to the p10 forecast. This would conceptually allow the p10 forecast to borrow strength from the p50 forecast, and I'd expect would be more likely to learn that p10 < p50 (though still not guaranteed).
